### PR TITLE
[SYCL][libdevice] Simplify SYCL libdevice cmake file.

### PR DESCRIPTION
**Pull request by jinge90**
Signed-off-by: jinge90 <ge.jin@intel.com>

### CODE REVIEW

**Comment by jinge90** [comment]
Hi, @intel/llvm-reviewers-runtime 
Could you review this PR? Previously, we had a very simple and not-well-organized cmakefile for libdevice since we have only a small number of files in libdevice. However, as we add more files and functions in libdevice(and we will add much more in the future), previous cmake file can't meet out requirement and will lead to a lot of code redundancy. This PR simplifies libdevice cmakefile by grouping same code into cmake functions.

Thanks very much.

File: libdevice/cmake/modules/SYCLLibdevice.cmake [hunk 1]
```diff
@@ -22,266 +25,83 @@ set(compile_opts
...
+add_devicelib_obj(libsycl-cmath-fp64 SRC cmath_wrapper_fp64.cpp DEP ${cmath_obj_deps} )
+add_devicelib_obj(libsycl-fallback-cmath SRC fallback-cmath.cpp DEP ${cmath_obj_deps})
+add_devicelib_obj(libsycl-fallback-cmath-fp64 SRC fallback-cmath-fp64.cpp DEP ${cmath_obj_deps})
+add_devicelib_spv(libsycl-fallback-cmath SRC fallback-cmath.cpp DEP ${cmath_obj_deps})
+add_devicelib_spv(libsycl-fallback-cmath-fp64 SRC fallback-cmath-fp64.cpp DEP ${cmath_obj_deps})
+
+set(itt_obj_deps device_itt.h spirv_vars.h device.h sycl-compiler)
+add_devicelib_obj(libsycl-itt-stubs SRC itt_stubs.cpp DEP ${itt_obj_deps})
+add_devicelib_obj(libsycl-itt-compiler-wrappers SRC itt_compiler_wrappers.cpp DEP ${itt_obj_deps})
+add_devicelib_obj(libsycl-itt-user-wrappers SRC itt_user_wrappers.cpp DEP ${itt_obj_deps})
```
Comment (aelovikov-intel) [inline libdevice/cmake/modules/SYCLLibdevice.cmake:107]: I think that having one extra function/macro/etc. that creates both spv and obj simultaneously will make this code easier to read. Can you please implement that?

Comment (jinge90) [reply libdevice/cmake/modules/SYCLLibdevice.cmake]: > I think that having one extra function/macro/etc. that creates both spv and obj simultaneously will make this code easier to read. Can you please implement that?

The function combining obj and spv creation is added and the grouping is updated too.

Thanks very much.

**Review by aelovikov-intel — COMMENTED** [review]
LGTM, but you still need an approve from someone with more rights in the project.

**Review by pvchupin — APPROVED** [review]

**Comment by pvchupin** [comment]
Fail is unrelated. Will be addressed in #6059.

**Pull request merged**